etcdserver/server.go

```diff
@@ -146,16 +146,20 @@ func NewServer(cfg *ServerConfig) (*EtcdServer, error) {
 	var n raft.Node
 	var s *raft.MemoryStorage
 	var id types.ID
-	walVersion, err := wal.DetectVersion(cfg.DataDir)
+	var ss *snap.Snapshotter
+	baseDataDir := cfg.DataDir
+	// Data dir lives under member now.
+	cfg.DataDir = path.Join(cfg.DataDir, "member")
```

Comment:
The root problem: What should be the data-dir in etcdserver.Config. 

Should it be the base dir (without member)? Or should it be the member dir? 
I am OK with both, but prefer member dir.

In my opinion whatever we choose, we should make it consistent throughout the code inside etcdserver pkg. In another word, cfg.DataDir should be readonly.